lenskit-eval/src/main/java/org/lenskit/eval/temporal/TemporalEvaluator.java

```diff
@@ -217,8 +250,42 @@ public void execute() throws IOException, RecommenderBuildException {
                 if (n > 0) {
                     rmse = sqrt(sse / n);
                 }
-                //writes the Prediction Score and TARMSE on file
-                tableWriter.writeRow(r.getUserId(), r.getItemId(), r.getValue(), r.getTimestamp(), predict, rmse);
+
+                //calculate recommendation rank
+                int rank = -1;
+                LongSet itemsInDao = new LongOpenHashSet(limitedDao.getItemIds().size());
+                itemsInDao.addAll(limitedDao.getItemIds());
+
+                if (itemsInDao.size() > 0) {
+                    //candidate =  r.getItemId() + random (items from dao - items by user, listsize-1);
+                    LongSet itemsByUser = new LongOpenHashSet(itemsInDao.size());
+
+                    if (limitedDao.getEventsForUser(r.getUserId()) != null) {
```

Comment:
Better to save the user's event list in a variable, so we just call `getEventsForUser` once.